contracts/DAOTreasury.cdc

```diff
@@ -7,6 +7,17 @@ pub contract DAOTreasury {
   pub let TreasuryStoragePath: StoragePath
   pub let TreasuryPublicPath: PublicPath
 
+  // Events
+  pub event TreasuryInitialized(initialSigners: [Address], initialThreshold: UInt64)
+  pub event ProposeAction(actionUUID: UInt64)
+  pub event ExecuteAction(actionUUID: UInt64)
+  pub event DepositVault(vaultID: String)
+  pub event DepositCollection(collectionID: String)
+  pub event WithdrawTokens(vaultID: String, amount: UFix64)
+  pub event WithdrawNFT(collectionID: String, nftID: UInt64)
```

Comment:
- can we add an event for a user signing a proposed action?
- can we add an event for an individual nft being sent to treasury?
- can we emit how many tokens were deposited in `DepositVault`?

Reply:
> - can we add an event for a user signing a proposed action?
Yes!

> can we add an event for an individual nft being sent to treasury?
> can we emit how many tokens were deposited in DepositVault?
Currently the way tokens are deposited into the treasury are by calling `borrowVaultPublic()`/`borrowCollectionPublic()`, and then calling the native `deposit` functions on those resource references.  Each token implementation is responsible for calling the `emit Deposit` events, or can choose not to, so it might be hard/inconsistent to capture those events if we rely on those native `Deposit` events.

In order to do this ourselves, we'd need to add specific `depositTokens` and `depositNFT` functions to DAOTreasury.cdc so we can emit our own events.

I'm definitely not against this, but it will need to be added after the Quantstamp audit.
Here is the ticket in Linear: https://linear.app/dappercollectives/issue/VSL-66/implement-depositnft-and-deposittokens-functions-for-treasury